pkg/volume/util/subpath/subpath_linux.go

```diff
@@ -110,9 +111,21 @@ func prepareSubpathTarget(mounter mount.Interface, subpath Subpath) (bool, strin
 		notMount = true
 	}
 	if !notMount {
-		// It's already mounted
-		klog.V(5).Infof("Skipping bind-mounting subpath %s: already mounted", bindPathTarget)
-		return true, bindPathTarget, nil
+		linuxHostUtil := hostutil.NewHostUtil()
+		mntInfo, err := linuxHostUtil.FindMountInfo(bindPathTarget)
+		if err != nil {
+			return false, "", fmt.Errorf("error calling findMountInfo for %s: %s", bindPathTarget, err)
+		}
+		if mntInfo.Root != subpath.Path {
```

Comment:
I found this not working correctly if configmap was not changed, mntInfo.Root is always not equal to subpath.Path when I debug here. Change subpath.Path to evalSubPath, this patch works well.
By the way, I backport this patch to v1.9.8 to test it